examples/rn-connection-and-error/frontend/app/providers/StoreProvider.tsx

```diff
@@ -62,19 +68,28 @@ export function StoreProvider({children}: PropsWithChildren) {
   const store = useObject(Store, user.customData.storeId);
   const products = useQuery(Product);
 
-  logger.info(`Users current store Id: ${user.customData.storeId}`);
+  // This is only used for logging when the store ID has changed.
+  // (Having `useEffect` rely on `user.customData.storeId` will not
+  // be sufficient in this case as the `storeId` is an `ObjectId`
+  // whose reference will change on each store update.)
+  const customDataStoreId = user.customData.storeId;
+  const storeIdChanged =
+    mostRecentStoreId?.toHexString() !== customDataStoreId.toHexString();
+  if (storeIdChanged) {
+    logger.info(`User's current store ID: ${customDataStoreId}`);
+    mostRecentStoreId = customDataStoreId;
+  }
 
   /**
-   * Adds a store. This demo app is syncing and using only 1 store with a
-   * specific store ID (see `app/atlas-app-services/config.ts`). Thus, if this
-   * store has already been created, this function will return immediately.
+   * Adds a store. This demo app is syncing only 1 store at a time. Thus, if
+   * a store has already been created, this function will return immediately.
    */
   const addStore = useCallback(() => {
     if (store) {
       return;
     }
     realm.write(() => {
-      realm.create(Store, {});
+      realm.create(Store, {_id: new BSON.ObjectId()});
```

Comment:
Since we updated to add the store on the backend, users probably won't call this function. Could be good to show though if a developer decides to do it differently. In this case, the `_id` here must be provided.

Reply:
We could also set this in the default method of `_id` in the model itself.

Reply:
But I'm actually leaning towards just removing the method if it's not used.

Reply:
Could definitely do that too 🙂 I'll push an update